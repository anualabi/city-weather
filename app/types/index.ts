export type CityData = {
  name: string;
  picture: string;
};

export type WeatherData = {
  date: string;
  city: CityData;
  tempType: string;
  temp: number;
};

export type IconName = "chevron-right";
