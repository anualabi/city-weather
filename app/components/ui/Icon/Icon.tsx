import { MaterialCommunityIcons } from "@expo/vector-icons";
import { Pressable, ViewStyle } from "react-native";

import { colors } from "@/theme";
import { IconName } from "@/types";

export interface IconProps {
  color?: string;
  onPress?: () => void;
  name: IconName;
  size?: number;
  testID?: string;
}

export default function Icon({
  color = colors.bgBlack,
  onPress,
  name,
  size = 40,
  testID,
}: IconProps) {
  const containerStyle: ViewStyle = {
    alignItems: "center",
    borderRadius: size / 2,
    height: size,
    justifyContent: "center",
    width: size,
  };

  return (
    <Pressable
      accessibilityRole="button"
      onPress={onPress}
      style={containerStyle}
    >
      <MaterialCommunityIcons
        color={color}
        name={name}
        size={size * 0.6}
        testID={testID ?? "icon"}
      />
    </Pressable>
  );
}
