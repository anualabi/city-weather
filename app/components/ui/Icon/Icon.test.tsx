import renderer from "react-test-renderer";
import "@testing-library/jest-native/extend-expect";
import { fireEvent, render } from "@testing-library/react-native";

import Icon, { IconProps } from "./Icon";
import { IconName } from "@/types";

const customRender = (props: IconProps) => {
  return render(<Icon {...props} />);
};

describe("Icon component", () => {
  test("should render correctly", () => {
    const tree = renderer.create(<Icon name="arrow-left" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should render the icon", () => {
    const { getByTestId } = customRender({
      name: "home" as IconName,
      color: "red",
      size: 40,
    });

    const icon = getByTestId("icon");
    expect(icon).toBeDefined();
  });

  test("should handle press events", () => {
    const onPress = jest.fn();
    const { getByRole } = customRender({
      name: "home" as IconName,
      onPress,
    });

    const button = getByRole("button");
    fireEvent.press(button);
    expect(onPress).toHaveBeenCalled();
  });
});
