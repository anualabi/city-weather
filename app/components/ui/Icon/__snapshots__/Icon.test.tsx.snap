// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component should render correctly 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderRadius": 20,
      "height": 40,
      "justifyContent": "center",
      "width": 40,
    }
  }
>
  <Text />
</View>
`;
